apps/api/lib/auth.js

```diff
@@ -4,22 +4,55 @@ import auth from 'express-basic-auth'
 import clownface from 'clownface'
 import $rdf from 'rdf-ext'
 import asyncMiddleware from 'middleware-async'
-import { DESCRIBE } from '@tpluscode/sparql-builder'
+import { DESCRIBE, INSERT, sparql } from '@tpluscode/sparql-builder'
 import { rdf, vcard } from '@tpluscode/rdf-ns-builders'
 import { isNamedNode } from 'is-graph-pointer'
+import onetime from 'onetime'
 
 const require = createRequire(import.meta.url)
 
 export function basic({ client }) {
+  const credentials = loadUsers()
+    .map(([name, { password }]) => ([name, password]))
+
   return Router()
     .use(auth({
-      users: loadUsers(),
+      users: Object.fromEntries(credentials),
       challenge: true,
       realm: 'view builder',
     }))
     .use(setAgent(client))
 }
 
+export function populateUsers({ client }) {
+  const insertUserData = onetime(async (req) => {
+    const users = loadUsers()
+
+    const query = users.reduce((previous, [id, { name }]) => {
+      const userId = req.rdf.namedNode(`/user/${id}`)
+
+      return sparql`${previous}
+      
+      DROP SILENT GRAPH ${userId};
```

Comment:
What happens if the users are deleted from the json file? would the become ghost users in the triplestore?

Reply:
Yes although the term "ghost" may not be all too accurate. A deleted user might be used as the `schema:author` of a view. 

If one is not used, would you rather have its triples deleted?

Reply:
You are right, the user is not to be deleted, but the access is, I think.

Reply:
Yes, that is right